forge-game/src/main/java/forge/game/cost/CostTap.java

```diff
@@ -66,7 +66,7 @@ public final boolean canPay(final SpellAbility ability, final Player payer, fina
 
     @Override
     public boolean payAsDecided(Player ai, PaymentDecision decision, SpellAbility ability, final boolean effect) {
-        ability.getHostCard().tap(true);
+        ability.getHostCard().tap(true, ability, ability.getHostCard().getController());
```

Comment:
imo this needs to be the Player param because of _Piracy_ corner case allowing you to pay opponents land costs